packages/amplication-data-service-generator/src/util/set-endpoint-permission.ts

```diff
@@ -0,0 +1,60 @@
+import assert from "assert";
+import { EnumEntityAction, EnumEntityPermissionType } from "../models";
+import { namedTypes } from "ast-types";
+import { Entity } from "../types";
+import { removeDecoratorByName, getClassMethodById } from "./ast";
+import { createPublicDecorator } from "./create-public-decorator";
+import { removeIdentifierFromUseInterceptorDecorator } from "./nestjs-code-generation";
+
+export const USE_ROLES_DECORATOR_NAME = "UseRoles";
+export const USE_INTERCEPTORS_DECORATOR_NAME = "UseInterceptors";
+export const PUBLIC_DECORATOR_NAME = "Public";
+
+const ACL_FILTER_RESPONSE_INTERCEPTOR_NAME = "AclFilterResponseInterceptor";
+const ACL_VALIDATE_REQUEST_INTERCEPTOR_NAME = "AclValidateRequestInterceptor";
+
+export function setEndpointPermissions(
+  classDeclaration: namedTypes.ClassDeclaration,
+  methodId: namedTypes.Identifier,
+  action: EnumEntityAction,
+  entity: Entity
+): void {
+  entity.permissions.find((entityPermission) => {
```

Comment:
I dont think this is what you meant to write... 
if you are using entity.permissions.find() then I would expect you just to find the element and exit, and only then use the element you found to continue with the logic